octodns/provider/yaml.py

```diff
@@ -248,10 +268,10 @@ def _apply(self, plan):
         self._do_apply(desired, data)
 
     def _do_apply(self, desired, data):
-        filename = join(self.directory, f'{desired.name}yaml')
+        filename = join(self.directory, f'{desired.decoded_name}yaml')
         self.log.debug('_apply:   writing filename=%s', filename)
         with open(filename, 'w') as fh:
-            safe_dump(dict(data), fh)
+            safe_dump(dict(data), fh, allow_unicode=True)
```

Comment:
This bits tells the yaml lib to write out utf-8 data rather than to do non-ascii chars as escaped hex.